### PR TITLE
[Merged by Bors] - chore(FieldTheory.Ratfunc): forward-port #19133

### LINKED ISSUE #4513: Fix FieldTheory.RatFunc
In the original file, sometimes `K` is a field and sometimes `K` is a
commutative ring (optionally, a domain). In some sections, this was
incorrectly translated to Lean 4 in #4293. This blocks #4512

### PR DESCRIPTION
**Pull request by adomani**
This PR re-ports `FieldTheory/RatFunc`, now that the mathlib3 version uses `section`s, after PR[#19133](https://github.com/leanprover-community/mathlib/pull/19133).

This closes [#4513](https://github.com/leanprover-community/mathlib4/issues/4513) and closes #4373.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/FieldTheory/RatFunc.lean [hunk 1]
```diff
@@ -16,20 +16,23 @@ import Mathlib.RingTheory.Polynomial.Content
 /-!
 # The field of rational functions
 
+> THIS FILE IS SYNCHRONIZED WITH MATHLIB4.
+> Any changes to this file require a corresponding PR to mathlib4.
+
```
Comment (eric-wieser) [inline Mathlib/FieldTheory/RatFunc.lean:21]: Let's not add this!

Comment (adomani) [reply Mathlib/FieldTheory/RatFunc.lean:21]: Oops!

Comment (adomani) [reply Mathlib/FieldTheory/RatFunc.lean]: Technically, what it says is correct, though...

File: Mathlib/FieldTheory/RatFunc.lean [hunk 2]
```diff
@@ -793,12 +798,13 @@ instance : Field (RatFunc K) :=
     mul_inv_cancel := fun _ => mul_inv_cancel
     zpow := zpowRec }
 
-end Field
-
 section IsFractionRing
 
-/-! ### `RatFunc` as field of fractions of `Polynomial` -/
+/-! ### `ratfunc` as field of fractions of `polynomial` -/
```
Comment (eric-wieser) [inline Mathlib/FieldTheory/RatFunc.lean:803]: Docstring needs fixing here

File: Mathlib/FieldTheory/RatFunc.lean [hunk 3]
```diff
@@ -781,9 +788,7 @@ theorem liftRingHom_injective [Nontrivial R] (φ : R[X] →+* L) (hφ : Function
 
 end LiftHom
 
-variable (K)
```
Comment (eric-wieser) [inline Mathlib/FieldTheory/RatFunc.lean:784]: Where did this go?

Comment (adomani) [reply Mathlib/FieldTheory/RatFunc.lean:784]: I also wondered about this: the binder of the variable `K` has been updated inside a `section`, so it did not affect the binder of the `K` that is in scope here.

File: Mathlib/FieldTheory/RatFunc.lean [hunk 4]
```diff
@@ -598,6 +604,7 @@ instance instCommRing : CommRing (RatFunc R) :=
 
 section LiftHom
 
+variable {K}
```
Comment (eric-wieser) [inline Mathlib/FieldTheory/RatFunc.lean:607]: The mathport output has `variable {K}` _before_ `section LiftHom`.

Comment (adomani) [reply Mathlib/FieldTheory/RatFunc.lean:607]: Should I then move the `variable {K}` where it was and also reinstate (if needed) `variable (K)` after the `end LiftHom`?

Comment (eric-wieser) [reply Mathlib/FieldTheory/RatFunc.lean:607]: You should aim to match the mathport output where possible; so yes, reinstate the variables lines you deleted and moved.

File: Mathlib/FieldTheory/RatFunc.lean (hunk 3 above)
Comment (eric-wieser) [reply Mathlib/FieldTheory/RatFunc.lean:784]: I think this is caused by the mis-port in the comment below.

File: Mathlib/FieldTheory/RatFunc.lean (hunk 2 above)
Comment (eric-wieser) [reply Mathlib/FieldTheory/RatFunc.lean:803]:
```suggestion
/-! ### `RatFunc` as field of fractions of `Polynomial` -/
```

File: Mathlib/FieldTheory/RatFunc.lean (hunk 4 above)
Comment (adomani) [reply Mathlib/FieldTheory/RatFunc.lean]: Ok, I re-instated the previous variable update and hence added back the re-update!

**Comment by urkud** [comment]
What means "settle" in the PR description? If this PR supersedes #4373, then you should either close #4373, or add "closes #4373" so that Github will close it automatically once this PR is merged.

**Comment by adomani** [comment]
@urkud, I did not know about the "auto-close": I updated the PR description to explicitly say "close #4373"!

**Review by eric-wieser — COMMENTED** [review]
I was struggling to diff this against mathport, so I pushed a series of commits to try and make it easier.

The diff can be viewed at https://github.com/leanprover-community/mathlib4/pull/4580/files/bf9631adcb77d2b68058f6421506979bfe28ff1d..51aaa759c7f2e4873b746bf66443b1c52957a3b0.

**Review by eric-wieser — APPROVED** [review]
bors d+

Feel free to merge once CI is happy.

I didn't review this as a full re-port, but verified that the diff against the latest mathport output has no more `include/omit` hacks, and all the `section`/`variable`s are in the right places.

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**